docs/reference/platforms.rst

```diff
@@ -0,0 +1,259 @@
+built and where they run.
+
+For more information on how to build charms for specific bases and
+architectures, see the :doc:`Platforms how-to </howto/build-guides/select-platforms>`
+page.
+
+Standard notation
+~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on:
+        - <my-arch>
+        - <your-arch>
+        ...
+       build-for:
+        - <our-arch>
+
+The platform name is an arbitrary string that describes the platform. The
+recommended platform name is the ``build-for`` arch.
+
+See :ref:`build-on <reference-build-on>` and
+:ref:`build-for <reference-build-for>` for information on the ``build-on``
+and ``build-for`` keywords.
+
+.. _reference-platforms-shorthand:
+
+Shorthand notation
+~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <arch>:
+
+Shorthand notation is a simple way to describe charms that build on and build
+for the same architecture. Shorthand notation requires the platform name to be
+a :ref:`valid debian architecture <reference-architectures>`.
+
+.. _reference-platforms-multi-base:
+
+Multi-base notation
+~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <platform-name>:
+       build-on:
+        - <some-base>:<some-arch>
+        - <some-base>:<another-arch>
+        - <another-base>:<some-arch>
+        - <another-base>:<another-arch>
+       build-for:
+        - <base>:<arch>
+
+Multi-base charms define a :ref:`base <reference-bases>` in each platform entry
+instead of the top-level ``base`` and ``build-base`` keywords. Within each
+platform entry, the base must be the same for all ``build-on`` and
+``build-for`` entries.
+
+The recommended platform name is ``<distribution>-<series>-<build-for-arch>``
+(for example, ``ubuntu-24.04-riscv64``).
+
+.. _reference-platforms-multi-base-shorthand:
+
+Multi-base shorthand notation
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. code-block:: yaml
+
+   platforms:
+     <base>:<arch>:
+
+Multi-base charms can also use shorthand notation when the platform builds on
+and builds for the same architecture. This shorthand notation is a YAML dictionary
+that contains the shorthand name as a key and a null value. The platform name must be a
+:ref:`valid base <reference-bases>` and
+:ref:`debian architecture <reference-architectures>` formatted as
+``<base>:<arch>``.
+
+.. _reference-build-on:
+
+``build-on``
+~~~~~~~~~~~~
+
+``build-on`` is a list of architectures and optional bases that describes the
+environments where the platform can build. Entries can be formatted either as only
+the architecture (``<arch>``), or as base and architecture (``<base>:<arch>``).
+
+Example ``build-on`` values with only architectures include:
+
+.. code:: yaml
+
+    build-on:
+      - amd64
+      - riscv64
+
+Example ``build-on`` values containing both a base and an architecture include:
+
+.. code:: yaml
+
+    build-on:
+      - ubuntu@24.04:riscv64
+      - ubuntu@22.04:amd64
```

Comment:
Similarly, this entire snippet would fail but we could split this into 2 discrete snippets:

```
.. code:: yaml
    build-on:
      - ubuntu@24.04:riscv64
```
```
.. code:: yaml
    build-on:
      - ubuntu@22.04:amd64